radionets/dl_framework/callbacks.py

```diff
@@ -9,6 +9,106 @@
 from pathlib import Path
 from fastcore.foundation import L
 import matplotlib.pyplot as plt
+from radionets.evaluation.utils import (
+    load_data,
+    get_images,
+    eval_model,
+    make_axes_nice,
+    check_vmin_vmax,
+)
+from radionets.evaluation.plotting import create_OrBu
+from radionets.dl_framework.utils import get_ifft_torch
+
+OrBu = create_OrBu()
+
+
+class CometCallback(Callback):
+    def __init__(self, name, test_data, plot_n_epochs, scale):
+        from comet_ml import Experiment
+
+        self.experiment = Experiment(project_name=name)
+        self.data_path = test_data
+        self.plot_epoch = plot_n_epochs
+        self.test_ds = load_data(
+            self.data_path,
+            mode="test",
+            fourier=True,
+            source_list=False,
+        )
+        self.scale = scale
+
+    def after_train(self):
+        self.experiment.log_metric(
+            "Train Loss",
+            self.recorder._train_mets.map(_maybe_item),
+            epoch=self.epoch + 1,
+        )
+
+    def after_validate(self):
+        self.experiment.log_metric(
+            "Validation Loss",
+            self.recorder._valid_mets.map(_maybe_item),
+            epoch=self.epoch + 1,
+        )
+
+    def plot_test_pred(self):
+        img_test, img_true = get_images(self.test_ds, 1, norm_path="none", rand=False)
+        model = self.model
+        with self.experiment.test():
+            with torch.no_grad():
+                pred = eval_model(img_test, model)
+
+        fig, ((ax1, ax2), (ax3, ax4)) = plt.subplots(2, 2, figsize=(16, 10))
+        lim_phase = check_vmin_vmax(img_true[0, 1])
```

Comment:
Why not set the max and min values to pi and -pi, respectively?

Reply:
I thought maybe the max could also be lower then pi in the true distribution

Reply:
Yes, that's true. But I am more of a fan of using the same scaled color bar for both plots and looking at the colored differences than looking at differences in the color bar. But that's just a personal preference I think.